### PR TITLE
xkcd command

### PR DESCRIPTION
**Pull request by Gilbert142**
**Please describe the changes this PR makes and why it should be merged:**

**What does this PR do:**
- [ ] This PR changes internal functions, modules and/or utilities
  - [ ] This PR modifies the Extension API
- [x] This PR modifies commands
  - [ ] This PR changes command functions
  - [ ] This PR changes metadata for the command, updated in `commands.js`
  - [x] This PR removes and/or adds commands
- [ ] This PR modifies Web processing and/or content
  - [ ] This PR modifies static/ejs content
  - [ ] This PR modifies request processing (endpoint functions)
  - [ ] This PR modifies paths to existing resources
- [ ] This PR **only** includes non-code changes, like changes to default configurations, README, typos, etc.

### CODE REVIEW

File: Commands/Public/xkcd.js [hunk 1]
```diff
@@ -0,0 +1,55 @@
+const fetch = require("chainfetch");
+
+module.exports = async ({ Constants: { Colors, Text, APIs, UserAgent } }, documents, msg) => {
+	if (msg.suffix) {
+		const comicData = await fetch.get(APIs.XKCD(msg.suffix)).set({ "User-Agent": UserAgent, Accepts: "application/json" }).onlyBody()
```
Comment (vladfrangu) [inline Commands/Public/xkcd.js]: Isn't it `accept` the header?

File: Commands/Public/xkcd.js [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+			.catch(() => null);
+		if (!comicData) {
+			msg.send({
+				embed: {
+					color: Colors.SOFT_ERR,
+					description: `Doh! An XKCD comic with ID ${msg.suffix} wasn't found.`,
```
Comment (vladfrangu) [inline Commands/Public/xkcd.js]: `An`?

File: Commands/Public/xkcd.js [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+					},
+					footer: {
+						text: `#${comicData.num}`,
+					},
+					timestamp: new Date(`${comicData.year}-${comicData.month}-${comicData.day}`),
+				},
+			});
+		}
+	} else {
+		const comicData = await fetch.get(APIs.XKCD()).set({ "User-Agent": UserAgent, Accepts: "application/json" }).onlyBody()
```
Comment (vladfrangu) [inline Commands/Public/xkcd.js]: Code dupe 😠

**Review by vladfrangu — CHANGES_REQUESTED** [review]

**Comment by Gilbert142** [comment]
Don't mind the sneaky Console.js edit

File: Commands/Public/xkcd.js [hunk 4]
```diff
@@ -1,14 +1,14 @@
...
+		.catch(() => null);
 	if (msg.suffix) {
-		const comicData = await fetch.get(APIs.XKCD(msg.suffix)).set({ "User-Agent": UserAgent, Accepts: "application/json" }).onlyBody()
-			.catch(() => null);
 		if (!comicData) {
 			msg.send({
 				embed: {
 					color: Colors.SOFT_ERR,
-					description: `Doh! An XKCD comic with ID ${msg.suffix} wasn't found.`,
+					description: `Doh! A XKCD comic with ID ${msg.suffix} wasn't found.`,
```
Comment (vladfrangu) [inline Commands/Public/xkcd.js]: Still code dupe :notlikemeow:
description: suffix ? '' : ''

**Review by vladfrangu — CHANGES_REQUESTED** [review]

File: Commands/Public/xkcd.js (hunk 2 above)
Comment (wasdennnoch) [reply Commands/Public/xkcd.js]: It's pronounced "ex kay cee dee", so yes, an.

File: Commands/Public/xkcd.js [hunk 5]
```diff
@@ -0,0 +1,49 @@
+const fetch = require("chainfetch");
+
+module.exports = async ({ Constants: { Colors, Text, APIs, UserAgent } }, documents, msg) => {
+	const comicData = await fetch.get(APIs.XKCD(msg.suffix ? msg.suffix : undefined)).set({ "User-Agent": UserAgent, Accept: "application/json" }).onlyBody()
+		.catch(() => null);
+
+	const notFound = () => msg.send({
+		embed: {
+			color: Colors.SOFT_ERR,
+			description: msg.suffix ? `Doh! A XKCD comic with ID ${msg.suffix} wasn't found.` : `Doh! The current XKCD comic couldn't be fetched.`,
```
Comment (wasdennnoch) [inline Commands/Public/xkcd.js]: Commenting this as a new review again :P
It's pronounced "ex kay cee dee", so it's `an`, not `a`. It's the sound that decides, not the letter. Unless you say it as a full word, in which case you should stop drinking.

**Review by wasdennnoch — CHANGES_REQUESTED** [review]

File: Commands/Public/xkcd.js (hunk 5 above)
Comment (Gilbert142) [reply Commands/Public/xkcd.js]: (Why do i keep blindly following vlad on english advice)

File: Commands/Public/xkcd.js [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+	} else {
+		msg.send({
+			embed: {
+				color: Colors.RESPONSE,
+				title: comicData.title,
+				image: {
+					url: comicData.img || comicData.alt,
+				},
+				footer: {
+					text: `Latest XKCD Comic | #${comicData.num}`,
```
Comment (vladfrangu) [inline Commands/Public/xkcd.js]: Code 👏 dupe

**Review by vladfrangu — CHANGES_REQUESTED** [review]

**Pull request merged**